spec/system/admin/reports_spec.rb

```diff
@@ -44,6 +44,19 @@
       click_button "Go"
       expect(page).to have_content "EMAIL FIRST NAME"
     end
+
+    it "displays a friendly timeout message" do
+      ActiveJob::Base.queue_adapter.perform_enqueued_jobs = false
+      login_as_admin_and_visit admin_report_path(
+        report_type: :customers, report_subtype: :mailing_list
+      )
+      expect_any_instance_of(Admin::ReportsController).to receive(:sleep).
+        and_raise(Rack::Timeout::RequestTimeoutException.new(nil))
```

Comment:
👍 This is a nice quick and direct method to test for a timeout.
It's not as real as simulating a real timeout, but at least it doesn't slow down our test runs unnecessarily.

Reply:
This functionality is also relevant for non-background processing. I'm guessing that's less practical to test though. 
Is it worth testing in a controller spec though?

I never really know where to draw the line. This seems like good enough coverage.

Reply:
I thought of different ways of using Timecop or reduce the timeout to a second but they all don't work with rack-timeout because:

* Rack::Timeout reads the configuration at boot time and you can't modify it later unless you dig into the guts of it and mock private methods which would be brittle.
* Rack::Timeout uses it's own monotonic timer as far as I could gather from looking at the source. So modifying the system's time doesn't help.

So if we don't want to wait 15 seconds for the timeout, we do have to raise the error ourselves.

> This functionality is also relevant for non-background processing. I'm guessing that's less practical to test though.

We would need to mock another part to raise an exception. Possible but I don't think it's worth it at the moment. The old behaviour is that the snail is displayed on a timeout. The same will happen if my code is buggy, just with a different response code. Low risk here. And this is only temporary. Soon we want to stop the waiting at 10 seconds, before the Rack timeout,